R/class-BinRange.R

```diff
@@ -0,0 +1,127 @@
+check_bin_range <- function(object) {
+    errors <- character()
+    start <- object@binStart
+    end <- object@binEnd
+    label <- object@binLabel
+    value <- object@value
+    
+    if (sum(is.na(c(start, end))) == 1 || sum(c(!length(start), !length(end))) == 1) {
+      msg <- "Must provide either both binStart and binEnd or neither."
+      errors <- c(errors, msg) 
+    }
+
+    if (any(c(length(start), length(end), length(label), length(value)) > 1)) {
+      msg <- "More than one value provided for one of binStart, binEnd, binLabel or value."
+      errors <- c(errors, msg)
+    } else {
+      if (!any(is.na(c(start, end))) && any(is.na(suppressWarnings(as.numeric(c(start, end)))))) {
+        msg <- "Must provide binStart and binEnds which are coercible to numeric."
+        errors <- c(errors, msg)
+      } else {
+        if (!any(is.na(c(start, end))) && as.numeric(start) > as.numeric(end)) {
+          msg <- "Provided binStart is after BinEnd."
+          errors <- c(errors, msg)
+        }
+      }
+
+      if (!length(label) || is.na(label)) {
+        msg <- "Must provide a binLabel."
+        errors <- c(errors, msg)
+      }
+    }
+
+    if (class(start) != class(end)) {
+      msg <- "Provided binStart and binEnd must be of the same base type (numeric, Date, Posix)."
+      errors <- c(errors, msg) 
+    }   
+
+    return(if (length(errors) == 0) TRUE else errors)
+}
+
+#' Bin Range
+#' 
+#' A class to define the start, end and label for a bin.
+#' Optionally, may include a value for that bin.
+#' If no start and end are provided, then assume the 'bin'
+#' represents a category.
+#' 
+#' @slot binStart A number, Date or Posix value
+#' @slot binEnd A number, Date or Posix value
+#' @slot binLabel A string
+#' @slot value
+#' 
+#' @name BinRange-class
+#' @rdname BinRange-class
+#' @export 
+BinRange <- setClass("BinRange", representation(
```

Comment:
I think it's just a naming issue.

It looks like @dmgaldi is in the process of making different type definitions for the overlay values (request and I presume also response?) for categorical, date and numeric: https://github.com/VEuPathDB/EdaDataService/pull/251/files

Reply:
i dont much care how the bins are made.. i just need an object that can say i have a label and maybe a value. but i think ill rename it Bin rather than BinRange, since that doesnt make sense for categoricals. unless someone has a better name than Bin?